Added Search and View all functionalities

finarugeanina opened the pull request:
- added search and view all functionalities
- added an extra loop for main menu so it will be displayed until the user selects Exit Application

finarugeanina committed:
add Search and View all functionalities

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -85,42 +82,48 @@ private static void addEmployee(Object employee, String message) {
+				+ employeesJsonFile + " document");
+	}
+
+	private static JSONArray readFromFile(String employeesJsonFile, JSONArray allEmployeesList) {
```
you don't need the second parameter, since you are returning it: JSONArray allEmployeesList

finarugeanina replied:
I deleted this method since now the project has a load file option

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -133,15 +136,43 @@ private static int chooseAnOptionFromMainMenu() {
+	}
+
+	private static void searchEmployee() {
+		System.out.println("Please type in the id of the employee you want to search: ");
```
this way of searching comes with a file reading; in other words, everytime you want to search something, you need to read the same content repetitively from a file. a more elegant solution would be to read only once from that file.

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -133,15 +136,43 @@ private static int chooseAnOptionFromMainMenu() {
-	
+
+	private static void viewAllEmployees() {
+		System.out.println("This is the list of all employees: ");
```
same comment from above here

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -133,15 +136,43 @@ private static int chooseAnOptionFromMainMenu() {
 	}
-	
+
+	private static void viewAllEmployees() {
+		System.out.println("This is the list of all employees: ");
+		String employeesJsonFile = "employees.json";
+		JSONArray allEmployeesList = new JSONArray();
+		allEmployeesList = readFromFile(employeesJsonFile, allEmployeesList);
+		for (Object emp : allEmployeesList) {
+			JSONObject employeeObject = (JSONObject) ((JSONObject) emp).get("employee");
+			System.out.println();
+			System.out.println(employeeObject.toString().replaceAll("[{\"}]", "").replaceAll("[,]", "\n"));
+		}
+	}
```
same here

finarugeanina replied:
I deleted this method since now the project has a load file option

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -133,15 +136,43 @@ private static int chooseAnOptionFromMainMenu() {
+		return numberSelected;
+	}
+
+	private static void searchEmployee() {
+		System.out.println("Please type in the id of the employee you want to search: ");
+		Scanner scan = new Scanner(System.in);
+		int idOfTheEmployee = scan.nextInt();
+		String employeesJsonFile = "employees.json";
+		JSONArray allEmployeesList = new JSONArray();
+		allEmployeesList = readFromFile(employeesJsonFile, allEmployeesList);
+		for (Object emp : allEmployeesList) {
+			JSONObject employeeObject = (JSONObject) ((JSONObject) emp).get("employee");
+			if (Integer.parseInt(String.valueOf(employeeObject.get("EmployeeId"))) == idOfTheEmployee) {
+				System.out.println(employeeObject.toString().replaceAll("[{\"}]", "").replaceAll("[,]", "\n"));
+			}
+		}
```
this way of searching comes with a file reading; in other words, everytime you want to search something, you need to read the same content repetitively from a file. a more elegant solution would be to read only once from that file.

rbutnariu replied:
that file you are reading from isn't guaranteed to exist when you want to search/view employees

finarugeanina replied:
I changed to a global and static JSONArray which is used in every method.

rbutnariu commented:
within the next commit, please include the document with the project description as well

rbutnariu commented:
some more general notes:
- we should include a "save" option, which should consist in creating the json file or overwriting it; we shouldn't write in file every contact we add; unless we want to save the employees, we don't want to save them automatically from one running session to another;
- similarly, we need a "load" option, which will populate some variables with the content that we read from a file.

finarugeanina committed:
added Save and Load option and change the validation for id

finarugeanina commented:
> within the next commit, please include the document with the project description as well

I added the project description.

finarugeanina commented:
> some more general notes:
> 
> * we should include a "save" option, which should consist in creating the json file or overwriting it; we shouldn't write in file every contact we add; unless we want to save the employees, we don't want to save them automatically from one running session to another;
> * similarly, we need a "load" option, which will populate some variables with the content that we read from a file.

I added the load and save methods. I also changed the way we validate the id since now we compare with the data from the list in which we add the employees and not the json file.

rbutnariu commented:
👍

finarugeanina committed:
change the list of employees from JSONArray to ArrayList

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -12,7 +14,7 @@
 
 public class EmployeesAdministrationApp {
-	public static JSONArray listOfEmployees = new JSONArray();
+	public static List<HashMap<String, Object>> listOfEmployees = new ArrayList<HashMap<String, Object>>();
```
this should be HashMap<String, String>, as we discussed, where the key should be the ID and the value should be a string containing all the employee information

finarugeanina replied:
I changed the list of employees from List to HashMap

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -62,7 +64,7 @@ private static void loadFile() {
 					JSONArray arrayFromJsonFile = (JSONArray) parser.parse(new FileReader(jsonFile));
 					for (Object emp : arrayFromJsonFile) {
-						listOfEmployees.add(emp);
+						listOfEmployees.add((HashMap<String, Object>) emp);
```
this should be changed

finarugeanina replied:
I changed the list of employees from List to HashMap

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -151,21 +157,16 @@ private static void addEmployee(Object employee, String message) {
-		employeeDetailsJson.put("employee", employeeDetails);
-		JSONObject employeeDetailsJsonObject = new JSONObject(employeeDetailsJson);
-		listOfEmployees.add(employeeDetailsJsonObject);
+		listOfEmployees.add((HashMap<String, Object>) employeeDetails);
```
this needs to be changed

finarugeanina replied:
I changed the list of employees from List to HashMap

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -151,21 +157,16 @@ private static void addEmployee(Object employee, String message) {
-			HashMap<String, Object> object = (HashMap<String, Object>) ((HashMap<String, Object>) emp).get("employee");
-			JSONObject obj = new JSONObject(object);
-			if (Integer.parseInt(String.valueOf(obj.get("EmployeeId"))) == idOfTheEmployee) {
+		for (HashMap<String, Object> emp : listOfEmployees) {
```
here you should iterate through hash map, not through a list of hashmaps

finarugeanina replied:
I changed the list of employees from List to HashMap

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -200,11 +200,9 @@ private static void searchEmployee() {
-				JSONObject obj = new JSONObject(object);
-				if (Integer.parseInt(String.valueOf(obj.get("EmployeeId"))) == idOfTheEmployee) {
-					System.out.println(obj.toString().replaceAll("[{\"}]", "").replace(",", "\n"));
+			for (HashMap<String, Object> emp : listOfEmployees) {
+				if (Integer.parseInt(String.valueOf(emp.get("EmployeeId"))) == idOfTheEmployee) {
+					System.out.println(emp);
```
this needs to be changed

finarugeanina replied:
this is changed now

finarugeanina committed:
changed the list of employees from list to HashMap

rbutnariu commented on src/EmployeesAdministrationApp.java:
```diff
@@ -89,9 +94,13 @@ private static void saveFile() {
+					Map.Entry<String, String> pair = it.next();
+					JSONObject obj = new JSONObject();
+					obj.put(pair.getKey(), pair.getValue());
+					array.add(obj);
```
where is this used?
